engines/scumm/he/script_v60he.cpp

```diff
@@ -304,6 +304,17 @@ void ScummEngine_v60he::o60_setState() {
 	int state = pop();
 	int obj = pop();
 
+	// WORKAROUND for bug #4776, knocker floats while door is open
+	// This hides the knocker when the closed door object is set to hide.
+	if (_game.id == GID_FREDDI && _currentRoom == 7) {
+		if (_game.heversion == 71 && obj == 35 && state == 0) {
+			putState(26, 0);
+		}
+		if (_game.heversion == 73 && obj == 52 && state == 0) {
+			putState(43, 0);
+		}
+	}
+
```

Comment:
So what happens when the door closes again? Is the knocker's state updated somewhere else?

Reply:
Yes, after the door is closed again, the knocker is also drawn again. The scripts have code for that.